internal/pkg/service/stream/storage/statistics/provider_test.go

```diff
@@ -29,41 +35,86 @@ type providerTestCase struct {
+	// Fixtures
+	projectID := keboola.ProjectID(123)
+	branchKey := key.BranchKey{ProjectID: projectID, BranchID: 456}
+	sourceKey := key.SourceKey{BranchKey: branchKey, SourceID: "my-source"}
+	sinkKey := key.SinkKey{SourceKey: sourceKey, SinkID: "my-sink"}
 
-	l2Config := statistics.NewConfig().Cache.L2
-	l2Cache, err := cache.NewL2Cache(d.Logger(), clk, l1Cache, l2Config)
-	require.NoError(t, err)
-	defer l2Cache.Stop()
+	// Get services
+	d, mocked := dependencies.NewMockedLocalStorageScope(t, commonDeps.WithClock(clk))
+	client := mocked.TestEtcdClient()
+	defRepo := d.DefinitionRepository()
+	storageRepo := d.StorageRepository()
+	fileRepo := storageRepo.File()
+	sliceRepo := storageRepo.Slice()
+	volumeRepo := storageRepo.Volume()
+	statsRepo := d.StatisticsRepository()
+
+	// Log etcd operations
+	var etcdLogs bytes.Buffer
+	rawClient := d.EtcdClient()
+	rawClient.KV = etcdlogger.KVLogWrapper(rawClient.KV, &etcdLogs, etcdlogger.WithMinimal())
+
+	// Register active volumes
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		session, err := concurrency.NewSession(client)
+		require.NoError(t, err)
+		defer func() { require.NoError(t, session.Close()) }()
+		test.RegisterWriterVolumes(t, ctx, volumeRepo, session, 1)
+	}
+
+	// Create branch, source, sink, file, slice
+	// -----------------------------------------------------------------------------------------------------------------
+	{
+		branch := test.NewBranch(branchKey)
+		require.NoError(t, defRepo.Branch().Create(&branch, clk.Now(), by).Do(ctx).Err())
+		source := test.NewSource(sourceKey)
+		require.NoError(t, defRepo.Source().Create(&source, clk.Now(), by, "Create source").Do(ctx).Err())
+		sink := test.NewSinkWithLocalStorage(sinkKey)
+		require.NoError(t, defRepo.Sink().Create(&sink, clk.Now(), by, "Create sink").Do(ctx).Err())
+	}
 
-	sliceKey1 := test.NewSliceKeyOpenedAt("2000-01-01T01:00:00.000Z")
-	sliceKey2 := test.NewSliceKeyOpenedAt("2000-01-01T02:00:00.000Z")
-	sliceKey3 := test.NewSliceKeyOpenedAt("2000-01-01T03:00:00.000Z")
+	// Create 2 more files/slices (3 totally)
+	// -----------------------------------------------------------------------------------------------------------------
+	var sliceKey1, sliceKey2, sliceKey3 model.SliceKey
+	{
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		clk.Add(time.Hour)
+		require.NoError(t, fileRepo.Rotate(sinkKey, clk.Now()).Do(ctx).Err())
+
+		slices, err := sliceRepo.ListIn(sinkKey).Do(ctx).All()
+		require.NoError(t, err)
+		require.Len(t, slices, 3)
+		sliceKey1 = slices[0].SliceKey
+		sliceKey2 = slices[1].SliceKey
+		sliceKey3 = slices[2].SliceKey
+	}
 
 	// Define test cases
 	cases := []providerTestCase{
 		{
 			Description: "Empty",
 			Assert: func(provider repository.Provider) {
-				stats, err := provider.SinkStats(ctx, sliceKey1.SinkKey)
+				stats, err := provider.SinkStats(ctx, sinkKey)
```

Comment:
I am right if I say that it is ok to use both `sinkKey` and `sliceKey1.SinkKey` it do not change the `stats` object received.

Reply:
yes it is equal